src/runapi.js

```diff
@@ -0,0 +1,61 @@
+const apiCommands = require("./tdx-api-commands");
+
+function objectToListMapper(obj) {
+  return Object.keys(obj).sort().map((element) => ([element, Number(element)]));
+}
+
+function stringifyArgumentValue(args, keyList) {
+  if (keyList.length > 1) {
+    const key = keyList.shift();
+    if (key in args) stringifyArgumentValue(args[key], keyList);
+  } else if (keyList.length === 1) {
+    const key = keyList.shift();
+    if (key in args) {
+      args[key] = (typeof args[key] === "number") ?
+        args[key].toString() :
+        args[key];
+    }
+  }
+}
```

Comment:
This function is a bit confusing to me.

As far as I'm aware, you're doing [lodash~update](https://lodash.com/docs/4.17.15#update) with stringify, e.g.

```js
function stringifyArgumentValue(args, keyList) {
  require("lodash").update(args, keyList, (val) => (typeof val === "number") ? val.toString() : val));
}
```

Since it might be overkill to get lodash just for one function, you could implement it yourself, e.g.

```suggestion
// copies lodash~update, e.g. updates the value in object given by path
function update(obj, path, updater) {
  const pathToLast = path.slice(0, -1);
  const lastPath = path[path.length - 1];
  const objToUpdate = pathToLast.reduce((pointer, currentPath) => {pointer[currentPath]}, obj);
  lastObj[lastPath] = updater(lastObj[lastPath]);
}

function stringifyArgumentValue(object, pathToVal) {
  update(object, pathToVal, (val) => (typeof val === "number") ? val.toString() : val));
}
```

Reply:
Good solution. But it is still recursive. If you know of a non recursive option I will add it.